bdb-ip.md

```diff
@@ -0,0 +1,1000 @@
+- [Changes that are additions to the exiting HTTP API of BDB
+  v0.11](https://github.com/bigchaindb/bigchaindb/milestone/9)
+
+After releasing BDB v1.0, breaking changes to the external APIs that are not:
+
+- show stopper-bug fixes; or
+- security considerations
+
+won't be permitted anymore. Additions to the APIs are however always allowed.
+
+To resolve the outstanding issues, we'll hence limit the scope of this document
+to the already existing endpoints.
+
+
+## First task
+
+Please check the two listed milestones above for the correctness of categorized
+issues.
+
+
+## Methodology
+
+This document will go address each of the issues in the [breaking
+changes milestone](https://github.com/bigchaindb/bigchaindb/milestone/11) and
+discuss possible solutions to them. Once there is a consensus on the solutions,
+we can implement them into BDB core (mainly HTTP API), py and js driver and
+the BDB CLI.
+
+All issues are ordered in a dependency graph to avoid backtracking in decision
+making and merge conflicts in implementation (root nodes do not have
+dependencies):
+
+- 0. What does "breaking change" mean
+    - 1. Inconsistent naming of short-hand ids
+    - 2. Inconsistency in using relative links
+        - 3. Consolidate / and /api/v1
+            - 4. Remove host and port from URLs in API and root endpoint
+        - 5. /outputs?unspents= returns unexpected results
+            - 6. A new /outputs endpoint
+- 7. /statuses?tx_id needs to return status invalid
+- 8. /transaction/ID needs status flag
+- 9. /transaction/id and /transaction?asset_id?operation=CREATE return same content
+- 10. Include block_id and transaction status in /transactions/id
+
+
+## Second task
+
+Please make sure the proposed dependency graph is correct.
+
+
+## Issues in milestone
+
+In this section, we'll go over each of the changes ordered by the above
+dependency tree.
+
+
+### [0. What does "breaking change" mean?](https://github.com/bigchaindb/bigchaindb/pull/1522#discussion_r120822313)
+
+@krish7919 wrote:
+
+```
+Suppose we change something in BigchainDB API, and make the corresponding
+changes to the (officially) supported drivers, can we claim that as long as the
+driver API (with the user programs) remain the same, the change is permissible.
+That might give us more leeway going forward if we come up with something
+unexpected.
+```
+
+
+#### Proposal 1
+
+Breaking changes means, every change provoking an incompatibility for the
+following components:
+
+- HTTP API
+- Events API
+- official Python driver
+- official Javascript driver
+
+
+#### Proposal 2
+
+Breaking changes means, every change provoking an incompatibility for the
+following components:
+
+- official Python driver
+- official Javascript driver
+
+Notice that compatibility is achieved here through officially supported
+drivers that update according to the breaking changes in the HTTP and Events
+API.
+
+
+#### Favorite Proposal: Proposal 1
```

Comment:
I also don't think Proposal 2 is enough. For instance if the data model returned from an endpoint changes the driver may not care or may be modified to work with the new data model but the user application consuming this data will break.